src/SprykerEco/Zed/PunchoutCatalogs/Business/Writer/PunchoutCatalogsWriter.php

```diff
@@ -62,14 +68,59 @@ public function updateConnection(PunchoutCatalogConnectionTransfer $punchoutCata
 
         $isSuccessful = $this->punchoutCatalogsEntityManager->updatePunchoutCatalogConnection($punchoutCatalogConnectionTransfer);
 
+        if (!$isSuccessful) {
+            return $this->createUnsuccessfulPunchoutCatalogResponseTransfer(static::MESSAGE_ERROR_DURING_CONNECTION_UPDATE);
+        }
+
+        return $this->storePassword($punchoutCatalogConnectionTransfer, static::MESSAGE_ERROR_DURING_CONNECTION_UPDATE);
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\PunchoutCatalogConnectionTransfer $punchoutCatalogConnectionTransfer
+     * @param string $errorMessage
+     *
+     * @return \Generated\Shared\Transfer\PunchoutCatalogResponseTransfer
+     */
+    protected function storePassword(PunchoutCatalogConnectionTransfer $punchoutCatalogConnectionTransfer, string $errorMessage): PunchoutCatalogResponseTransfer
+    {
+        if (!$punchoutCatalogConnectionTransfer->getPassword()) {
+            return $this->createSuccessfulResponseTransfer($punchoutCatalogConnectionTransfer);
```

Comment:
```suggestion
            return $this-> createUnsuccessfulPunchoutCatalogResponseTransfer($punchoutCatalogConnectionTransfer);
```

Reply:
Nope, if no password response should be successful, to keep functionality works.
But I agree that method could be factored, please let me know if you have any suggestions